internal/cmd/marketplace/delete.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Mia srl
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package marketplace
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/mia-platform/miactl/internal/client"
+	"github.com/mia-platform/miactl/internal/clioptions"
+	"github.com/spf13/cobra"
+)
+
+const (
+	deleteMarketplaceEndpoint = "/api/backend/marketplace/tenants/%s/resources/%s"
+)
+
+// DeleteCmd return a new cobra command for deleting a single marketplace resource
+func DeleteCmd(options *clioptions.CLIOptions) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "delete [resource-id]",
+		Short: "Delete marketplace item",
+		Long:  `Delete a single marketplace item by its ID`,
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			restConfig, err := options.ToRESTConfig()
+			cobra.CheckErr(err)
+			client, err := client.APIClientForConfig(restConfig)
+			cobra.CheckErr(err)
+
+			var resourceID string
+			if len(args) > 0 {
+				resourceID = args[0]
+			}
```

Comment:
```suggestion
			resourceID := args[0]
```
`Args:  cobra.ExactArgs(1),` should guarantee that exactly one exists

Reply:
done